src/main/java/pl/zaprogramujzycie/mma/controllers/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import pl.zaprogramujzycie.mma.utils.exceptions.ErrorMessage;
+
+import java.util.Date;
+
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(ChangeSetPersister.NotFoundException.class)
+    public ResponseEntity<ErrorMessage> resourceNotFoundException(ChangeSetPersister.NotFoundException ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.NOT_FOUND.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+        return new ResponseEntity<>(message, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(HttpClientErrorException.Unauthorized.class)
+    public ResponseEntity<ErrorMessage> unauthorizedException (HttpClientErrorException.Unauthorized ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.UNAUTHORIZED.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+
+        return  new ResponseEntity<>(message, HttpStatus.UNAUTHORIZED);
+    }
+
+    @ExceptionHandler(HttpClientErrorException.Forbidden.class)
+    public ResponseEntity<ErrorMessage> unauthorizedException (HttpClientErrorException.Forbidden ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.FORBIDDEN.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+
+        return  new ResponseEntity<>(message, HttpStatus.FORBIDDEN);
+    }
+
+
+    @ExceptionHandler(HttpClientErrorException.BadRequest.class)
+    public ResponseEntity<ErrorMessage> badRequestException(HttpClientErrorException.BadRequest ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.BAD_REQUEST.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+
+        return  new ResponseEntity<>(message, HttpStatus.BAD_REQUEST);
+    }
+
+
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    public ResponseEntity<ErrorMessage> dataIntegrityViolationException(DataIntegrityViolationException ex, WebRequest request) {
+        ErrorMessage message = new ErrorMessage(
+                HttpStatus.IM_USED.value(),
+                new Date(),
+                ex.getMessage(),
+                request.getDescription(false));
+
+        return  new ResponseEntity<>(message, HttpStatus.IM_USED);
+    }
+
+    @ExceptionHandler(Exception.class)
```

Comment:
Testowalas to? Bo kazdy exception dziedziczy po Exception, nie jest czasem tak ze ten handler lapie wszystkie wyjatki?

Reply:
Czytałam trochę, ale to jest kolejny większy kawałek wiedzy, którego mi brakuje i uznałam, że puki działa na h2 to zajmę się budowaniem tej logiki. I bez Testcontainers cały czas muszę doczytywać i kilkukrotnie więcej czasu spędzam na nauce niż pisaniu kodu.